lib/active_merchant/billing/gateways/orbital.rb

```diff
@@ -605,6 +662,27 @@ def add_managed_billing(xml, options)
         end
       end
 
+      # Validates Electronic Check Serial number based on bin value provided
+      def validate_ecp_check_serial(serial_number, bin)
+        (bin.eql?('000001') && serial_number.length.eql?(9)) || (bin.eql?('000002') && serial_number.length.eql?(6))
+      end
+
+      # Adds ECP conditional attributes depending on other attribute values
+      def add_ecp_details(xml, parameters = {})
+        requires!(parameters, :check_serial_number) if parameters[:auth_method]&.eql?('A') || parameters[:auth_method]&.eql?('P')
+        xml.tag! :ECPActionCode, parameters[:action_code] if parameters[:action_code] && ECP_ACTION_CODES.include?(parameters[:action_code])
+        if parameters[:auth_method]&.eql?('A') || parameters[:auth_method]&.eql?('P')
+          raise 'Invalid Check Serial Number' unless validate_ecp_check_serial(parameters[:check_serial_number], bin)
```

Comment:
Same as comment on line 527.

Reply:
Reason is similar to the previous comment.

Reply:
We should remove the `raise` flag and use conditionals where appropriate, let the gateway process the transaction and handle responses accordingly.

Reply:
Acknowledged.